allensdk/api/queries/brain_observatory_api.py

```diff
@@ -346,6 +355,27 @@ def save_ophys_experiment_data(self, ophys_experiment_id, file_name):
 
         self.retrieve_file_over_http(self.api_url + file_url, file_name)
 
+    @cacheable(strategy='create',
+               pathfinder=Cache.pathfinder(file_name_position=2,
+                                           path_keyword='file_name'))
+    def save_ophys_experiment_analysis_data(self, ophys_experiment_id, file_name):
+
+        data = self.template_query('brain_observatory_queries',
+                                   'ophys_analysis_file',
+                                   ophys_experiment_id=ophys_experiment_id)
+
+        try:
+            file_url = data[0]['download_link']
+        except Exception as _:
+            raise Exception("ophys experiment %d has no %s analysis file" %
```

Comment:
@JFPerkins caught this is A) busted, and B) should be more specific with its exception catching for either an IndexError or KeyError; needs to be changed other places as well, and perhaps a test that exercises the case of a bad piece of data coming from the query